homeassistant/components/keymitt_ble/__init__.py

```diff
@@ -0,0 +1,135 @@
+"""Integration to integrate MicroBot with Home Assistant."""
+from __future__ import annotations
+
+import asyncio
+import contextlib
+import logging
+from typing import TYPE_CHECKING, Any
+import async_timeout
+
+from microbot import (
+    MicroBotApiClient,
+    parse_advertisement_data,
+)
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth.passive_update_coordinator import (
+    PassiveBluetoothDataUpdateCoordinator,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, ServiceCall, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.const import CONF_ACCESS_TOKEN
+
+from .const import (
+    CONF_BDADDR,
+    DEFAULT_RETRY_COUNT,
+    DOMAIN,
+)
+
+if TYPE_CHECKING:
+    from bleak.backends.device import BLEDevice
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+PLATFORMS: list[str] = ["switch"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
+    hass.data.setdefault(DOMAIN, {})
+    token: Any | None = entry.data.get(CONF_ACCESS_TOKEN)
+    bdaddr: Any | None = entry.data.get(CONF_BDADDR)
+    assert bdaddr is not None
+    assert token is not None
+    ble_device: BLEDevice | None = bluetooth.async_ble_device_from_address(
+        hass, bdaddr.upper()
```

Comment:
It would be better to store the original address you get back from bleak / bluetooth service info instead of mutating it to uppercase